testah-junit/src/main/java/org/testah/framework/cli/TestFilter.java

```diff
@@ -165,62 +152,41 @@ public int loadCompiledTestClass(final String internalClass) {
-                                TS.params().getFilterByTestType() + "]");
-                        continue;
-                    }
-                }
-                if (filterByTag) {
-                    filter = filterParams.getFilterByTag();
-                    if (!isFilterCheckOk(meta.tags(), filter)) {
-                        TS.log().trace("test[" + test.getName() + "] filtered out by getFilterByTag");
-                        continue;
-                    }
-                }
-                if (filterByComponent) {
-                    filter = filterParams.getFilterByComponent();
-                    if (!isFilterCheckOk(meta.components(), filter)) {
-                        TS.log().trace("test[" + test.getName() + "] filtered out by getFilterByComponent");
-                        continue;
-                    }
-                }
-                if (filterByDevice) {
-                    filter = filterParams.getFilterByDevice();
-                    if (!isFilterCheckOk(meta.devices(), filter)) {
-                        TS.log().trace("test[" + test.getName() + "] filtered out by getFilterByDevice");
-                        continue;
-                    }
-                }
-                if (filterByPlatform) {
-                    filter = filterParams.getFilterByPlatform();
-                    if (!isFilterCheckOk(meta.platforms(), filter)) {
-                        TS.log().trace("test[" + test.getName() + "] filtered out by getFilterByPlatform");
-                        continue;
-                    }
-                }
-                if (filterByRunType) {
-                    filter = filterParams.getFilterByRunType();
-                    if (!isFilterCheckOk(meta.runTypes(), filter)) {
-                        TS.log().trace("test[" + test.getName() + "] filtered out by getFilterByRunType");
-                        continue;
+
+                List<TestCaseDto> testCases = meta.getTestCases();
+                boolean atleastOneTestMeetsCriteria = false;
+                if (testCases.size() > 0) {
+                    for (TestCaseDto testCase : testCases) {
+                        if (filterTestCase(testCase, testCase.getName())) {
+                            atleastOneTestMeetsCriteria = true;
+                            break;
+                        }
                     }
                 }
 
-                testClassesMetFiltersToUse.add(test);
+                if (atleastOneTestMeetsCriteria) {
+                    testClassesMetFiltersToUse.add(test);
+                } else {
+                    TS.log().trace(String.format("No TestCases would run for testPlan[%s], if attributes are applied at the " +
+                            "testplan level, they will not be applied at the " +
+                            "testcase level if they too supply values for thoses attributes.", meta.name()));
```

Comment:
thoses?

Actually I am not sure what you are trying to say. Is it:
If the attributes from test plan and test case level are applied no test cases will be run. Therefore, the test case attributes will be ignored.
?